### PR TITLE
Creating cluster configuration document

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

**Comment by omertuc** [comment]
I believe I've added a lot of this documentation inline within the structs in https://github.com/openshift-kni/lifecycle-agent/pull/223 , so this PR may be redundant, wdyt?

File: docs/post-pivot-configuration.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// TODO: describe folder structure
+Folder structures can be seen https://github.com/openshift-kni/lifecycle-agent/blob/main/ibu-imager/installation_configuration_files/scripts/prepare-installation-configuration.sh#L30 
+and will be described later on
+
+## Network configuration
+
+User can provide specific network configuration that he wants to be applied on the node as part of installation process.
+Currently, we support nmconnection files that should be placed in /opt/openshift/network-configuration/system-connections
+Prepare installation service is on charge to copy them to network manager configuration folder and restart NM
+https://github.com/openshift-kni/lifecycle-agent/blob/main/ibu-imager/installation_configuration_files/scripts/prepare-installation-configuration.sh#L54
```
Comment (donpenney) [inline docs/post-pivot-configuration.md:35]: nit: I think these links could be relative paths, which would avoid the explicit repo and branch reference. ie.
`[prepare-installation-configuration.sh](installation_configuration_files/scripts/prepare-installation-configuration.sh#L54)`

File: docs/post-pivot-configuration.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+- [IBU/I post pivot configuration](#ibi-post-pivot-configuration)
+  - [Overview](#overview)
+  - [Network configuration](#network-configuration)
+  - [Recertification flow](#recertification-flow)
+  - [User specifications](#user-specifications)
+
+## Overview
+
+Post pivot configuration is on charge for applying user configurations after rebooting in new stateroot.
```
Comment (eranco74) [inline docs/post-pivot-configuration.md:11]:
```suggestion
Post pivot configuration is in charge of applying user configurations after rebooting into the new stateroot.
```

File: docs/post-pivot-configuration.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- Perform networking configuration
+- Recertification flow
+- User specifications:
+  - Run recert flow
```
Comment (eranco74) [inline docs/post-pivot-configuration.md:15]: This is repeating line 13 no?

File: docs/post-pivot-configuration.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  - Apply new hostname
+  - Apply new ip
+  - Apply all user specific configurations that user provides as https://github.com/openshift-kni/lifecycle-agent/blob/main/ibu-imager/clusterinfo/clusterinfo.go#L8
```
Comment (eranco74) [inline docs/post-pivot-configuration.md:18]: @omertuc please update this doc with the proper link once https://github.com/openshift-kni/lifecycle-agent/pull/223 is merged

Comment (eranco74) [reply docs/post-pivot-configuration.md:18]: Isn't the new IP and hostname part of the configuration the user provides here?

File: docs/post-pivot-configuration.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  - Applying extra manifests
+
+## Mount configuration folder
+There are 2 differences ways to provide configuration folder to:
```
Comment (eranco74) [inline docs/post-pivot-configuration.md:22]:
```suggestion
There are 2 differenet ways to provide the configuration folder:
```

File: docs/post-pivot-configuration.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+1. Add it as part of stateroot configuration process as we do in IBU, should be placed in /opt/openshift 
+2. Provide it as block device with label "relocation-config" , in this case we will mount it.
```
Comment (eranco74) [inline docs/post-pivot-configuration.md:24]: I'll open a task to use a better label

File: docs/post-pivot-configuration.md (hunk 1 above)
Comment (eranco74) [reply docs/post-pivot-configuration.md:35]: Also, please use the markdown syntax as Don suggested
[Link text](URL/relative path)

**Comment by eranco74** [comment]
OK, let's merge this and add the new configuration options in another PR
@tsorya :point_up: 
/approve
/lgtm

**Pull request merged**